ns/crypto.n3

```diff
@@ -0,0 +1,94 @@
+#  Schema for crypto built-in functions in cwm. 
+#
+#  These are terms drawn to match the available functions
+# in mxCrypto. See:
+# http://www.amk.ca/python/writing/pycrypt/  for the manual and
+# http://www.amk.ca/python/code/crypto.html  for the code.
+#
+#  I am not sure whether in the domain and range functions we should
+# distinguish strings one can present to a person from crypto binary.
+# Might be useful for driving the UI and/or serialization using base64 etc.
+#
+#  Beware that the simplicity of use of some of these properties belies
+# the knowledge o fcrypography that one needs to make a secure system
+#
+@prefix rdf:  <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix owl: <http://www.w3.org/2002/07/owl#>.
+@prefix log:  <http://www.w3.org/2000/10/swap/log#> .
+
+@prefix string:  <http://www.w3.org/2000/10/swap/string#> .
+@prefix cr:  <http://www.w3.org/2000/10/swap/crypto#> .
+
+
+#####################################################################
+#
+# Hash functions
+
+cr:HashFunction a rdfs:Class;
+    rdfs:label "Any crypographic hash function";
+    rdfs:comment """
+The crypographic hash functions are (being functions) unique and are,
+when secure, assumed unambiguous (the whole point of being hash
+functions). That is, when you have the right hash, you have the right
+document. Currently (2001/9) only SHA is given that property.
+""".
+
+cr:md5 a rdf:Property, owl:FunctionalProperty, owl:InverseFunctionalProperty;
+    rdfs:label "MD5 hash";
+    rdfs:comment "The object is a MD5 hash of the subject.";
+    rdfs:domain string:String;
+    rdfs:range string:String.
+
+cr:sha a rdf:Property, owl:FunctionalProperty, owl:InverseFunctionalProperty;
+    rdfs:label "SHA hash";
+    rdfs:comment "The object is a SHA-1 hash of the subject.";
+    rdfs:domain string:String;
+    rdfs:range string:String.
+
+###########################  Public Key Algorithms
```

Comment:
Do we use these at all?

Reply:
Unsure whether they are currently being used ([this CSV](https://github.com/w3c/N3/blob/master/useOfBuiltins/results.csv) seems to imply that they're rarely used)

But, at least in the builtin wishlist there is a request for SHA.